src/lightning/pytorch/loggers/mlflow.py

```diff
@@ -100,6 +101,7 @@ def any_lightning_module_function_or_hook(self):
         artifact_location: The location to store run artifacts. If not provided, the server picks an appropriate
             default.
         run_id: The run identifier of the experiment. If not provided, a new run is started.
+        synchronous: Which value to pass for the ``synchronous`` flag where applicable. Requires mlflow >= 2.8.0
```

Comment:
I suggest to explain in plain words what it does. This way, it's easier to discover for users who may not also look at the mlflow docs at the same time.

```suggestion
        synchronous: Whether to block the execution every logging call until complete. Defaults to True (blocking) if unspecified.
            Setting to False requires mlflow >= 2.8.0.
```

Reply:
@awaelchli I change for sure, but we can't guarantee this will be how it works - we are leaving this up to `mlflow` that can simply ignore this flag (potentially). The default is actually - up to `mlflow` too.
Had we used our own implementation with the thread pool we could have guaranteed this.

How about `Hints mlflow whether to block the execution for every logging call until complete where applicable.`? I'd like to keep `where applicable` since `log_artifacts` does not have this options, it's always synchronous.

Reply:
Yes sure, feel free to add that part back. My main ask is just to explain it plainly and briefly so it's clear.